src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -494,10 +524,13 @@ public string GenerateResponseFileContents()
         /// <summary>
         /// Get the command line arguments to pass to the compiler.
         /// </summary>
-        private string[] GetArguments(string commandLineCommands, string responseFileCommands)
+        private string[] GetArguments(string commandLineCommands, string responseFileCommands, bool emitServerLog)
```

Comment:
This extra arg was required to avoid logging to the server a compilation step that would never occur.
It comes from mixing logging into a method simply called GetArguments. Arguably, the logging code should be moved to the caller that wants it.